### PR TITLE
Clang-format c/cpp files in test directory

### PR DESCRIPTION
**Pull request by aheejin**
Context: https://github.com/WebAssembly/binaryen/pull/4188#discussion_r717075686

### CODE REVIEW

**Review by kripken — APPROVED** [review]

**Comment by aheejin** [comment]
I reverted bigswitch.cpp changes and excluded the file from clang-format by adding
```cpp
// clang-format off
```
on the top.

There are big commented-out code blocks, and apparently clang-format messes up formatting in them. Also to make matters worse, different clang-format versions do different things on those commented-out code blocks, which caused the previous CI failure.

File: test/bigswitch.cpp [hunk 1]
```diff
@@ -1,7934 +1,9627 @@
-#include <stdio.h>
```
[the remark this answers is not in the document]
Comment (tlively) [reply test/bigswitch.cpp:1]: Already done!

**Review by tlively — APPROVED** [review]

File: test/control_flow.cpp [hunk 2]
```diff
@@ -1,23 +1,26 @@
-#include <cmath>
 #include <algorithm>
+#include <cmath>
 #include <emscripten.h>
 
 extern "C" {
 
 int EMSCRIPTEN_KEEPALIVE check_if(int x) {
-  if (x < 10) x++;
+  if (x < 10)
```
[the remark this answers is not in the document]
Comment (aheejin) [reply test/control_flow.cpp:8]: That's the area of clang-tidy. (clang-format doesn't add or remove chars)
I think making test/ clang-tidy compliant is doable but a harder task.

File: test/bigswitch.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (aheejin) [reply test/bigswitch.cpp:1]: https://github.com/WebAssembly/binaryen/pull/4192#issuecomment-929689269

**Comment by aheejin** [comment]
What are these *.cpp files within test/ used for? I know c/cpp within example/ is used for gcc tests but not sure if cpp files under test/ are used at all? Some of them are causing clang-tidy errors. One reason is missing braces, and the other reason is inability to find the header like `emscripten.h`: https://github.com/WebAssembly/binaryen/pull/4192/checks?check_run_id=3738423726

This makes me wonder whether these files are even used and what these files are for.

**Comment by kripken** [comment]
I think the cpp files there were used to generate the `.wat` files in the past, so they could be used to recreate them. But they might have little value now - it's been a very long time since we've used them.

**Pull request merged**